src/heronarts/lx/parameter/BoundedFunctionalParameter.java

```diff
@@ -0,0 +1,121 @@
+package heronarts.lx.parameter;
+
+import heronarts.lx.parameter.BoundedParameter.Range;
+
+/**
+ * A FunctionalParameter that is bounded by a range.  This enables the calculation
+ * of normalized values which enables mapping with the modulation engine.
+ */
+public abstract class BoundedFunctionalParameter extends FunctionalParameter implements LXNormalizedParameter {
+
+  /**
+   * Range of the parameter
+   */
+  public final Range range;
+
+  /**
+   * A bounded functional parameter with a range of 0-1
+   *
+   * @param label Label for parameter
+   */
+  public BoundedFunctionalParameter(String label) {
+    this(label, 0, 1);
+  }
+
+  /**
+   * A bounded functional parameter with a range of 0 to max
+   *
+   * @param label Label for parameter
+   * @param max Maximum value
+   */
+  protected BoundedFunctionalParameter(String label, double max) {
+    this(label, 0, max);
+  }
+
+  /**
+   * A bounded functional parameter with a range from v0 to v1. Note that it is not necessary for
+   * v0 to be less than v1, if it is desired for the knob's value to progress negatively.
+   *
+   * @param label Label for parameter
+   * @param v0 Start of range
+   * @param v1 End of range
+   */
+  protected BoundedFunctionalParameter(String label, double v0, double v1) {
+    super(label);
+    this.range = new Range(v0, v1, this);
+  }
+
+  /**
+   * Not supported for this parameter type unless subclass overrides.
+   *
+   * @param value The value
+   */
+  @Override
+  public LXParameter setValue(double value) {
+    throw new UnsupportedOperationException(
+        "BoundedFunctionalParameter does not support setValue()");
+  }
+
+  /**
+   * Retrieves the raw value of the parameter, subclass must implement.
+   * This value will be constrained to the range.
+   *
+   * @return Parameter value
+   */
+  protected abstract double getValueInternal();
```

Comment:
Going to rename this to computeValue() which I think will be a little more clear

Reply:
Great.